File: src/main/java/com/j148/backend/hearing/model/Hearing.java
package com.j148.backend.hearing.model;
import com.j148.backend.contractor.model.Contractor;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Hearing {
    @Min(value = 1, message = "Hearing ID cannot be less than 1")
    @NotNull(message = "Hearing ID cannot be Null")
    private Long hearingsId;
    private LocalDateTime scheduleDate;
    private Outcome outcome;
    @NotBlank(message = "A reason for a Hearing is required")
    @Size(min = 10 , max = 40 , message = "Reason description is too short or too long (10 Min - 40 max)")
    private String reason;
    private Contractor contractor;

    public enum Outcome {
        NULL, SUSPENDED, CLEARED
    }
}

File: hrmsPython/models/Hearing.py
from dataclasses import dataclass
from datetime import datetime
from enum import Enum
from typing import Optional

from dataclasses_json import dataclass_json

from models import Contractor  # Assuming Contractor is in the 'models' module


class Outcome(Enum):
    NULL = "NULL"
    SUSPENDED = "SUSPENDED"
    CLEARED = "CLEARED"

@dataclass_json
@dataclass
class Hearing:
    hearingsId: Optional[int] = None
    scheduleDate: Optional[datetime] = None
    outcome: Optional[Outcome] = None
    reason: Optional[str] = None
    contractor: Optional[Contractor] = None
